src/spikeinterface/core/sortinganalyzer.py

```diff
@@ -1158,6 +1163,55 @@ def get_default_extension_params(self, extension_name: str):
         return get_default_analyzer_extension_params(extension_name)
 
 
+def _sort_extensions_by_dependency(extensions):
+    """
+    Sorts a dictionary of extensions so that the parents of each extension are on the "left" of their children.
+    Assumes there is a valid ordering of the included extensions.
+
+    Parameters
+    ----------
+    extensions: dict
+        A dict of extensions.
+
+    Returns
+    -------
+    sorted_extensions: dict
+        A dict of extensions, with the parents on the left of their children.
+    """
+
+    extensions_list = list(extensions.keys())
+    extension_params = list(extensions.values())
+
+    i = 0
+    while i < len(extensions_list):
+
+        extension = extensions_list[i]
+        dependencies = get_extension_class(extension).depend_on
```

Comment:
I am not sure you are handling the case with "|" that is either this dep either this dep
like ComputeTemplates.depend_on = ["random_spikes|waveforms"]

Reply:
Yes, you are right. This failure meant that these weren't being re-ordered correctly. I've now updated by flattening out e.g. ["templates", "random_spikes|waveforms"] into ["templates", "random_spikes", "waveforms"] when we first read the dependencies of an extension.

Reply:
cool, this should make the trick. thanks